frontend/src/app/crm/leads/leads.component.ts

```diff
@@ -31,31 +49,88 @@ export class LeadsComponent implements AfterViewInit {
 
   constructor(private readonly leadService: LeadService) {
     this.dataSource = new MatTableDataSource();
-
     this.loadAllLeads();
   }
 
   ngAfterViewInit(): void {
     this.dataSource.paginator = this.paginator;
+
+    /**
+     * This will let the dataSource sort feature to access nested properties in the JSON such as column_data.
+     */
+    this.dataSource.sortingDataAccessor = (lead, property) => {
+      switch(property) {
+        case 'name': return lead.column_data.FULL_NAME;
+        case 'phone_number': return lead.column_data.PHONE_NUMBER;
+        default: return lead[property];
+      }
+    };
     this.dataSource.sort = this.sort;
   }
 
+ /**
+  * This will access the leads from the leadService and handle the filterPredicate and the isLoading boolean value.
+  */
   loadAllLeads(): void {
     this.isLoading$.next(true);
     this.leadService.getAllLeads().pipe(first()).subscribe((leads) => {
-      this.dataSource.data = leads;
-      this.isLoading$.next(false);
+    this.dataSource.data = leads;
+
+    /**
+     * @param data The whole data we have in the JSON.
+     * @param filter The value that the user searches transformedFilter.
+     */
+    this.dataSource.filterPredicate = (data, filter: string)  => {
+      const accumulator = (currentTerm, key) => {
+        //Using the recusrsive method here becuase using an if/else condition will increase space complexity n times.
+        return this.nestedPropertyFilterCheck(currentTerm, data, key);
+        console.log('1 abcd  ' + currentTerm );
```

Comment:
prefer to delete debugging logs or give them a significant message so you can remember what it is about.

Reply:
Sure. I don't actually need it any more.